build-tools/packages/build-cli/src/commands/check/latestVersions.ts

```diff
@@ -0,0 +1,66 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Args } from "@oclif/core";
+import { findPackageOrReleaseGroup, packageOrReleaseGroupArg } from "../../args.js";
+import { BaseCommand, sortVersions } from "../../library/index.js";
+
+export default class LatestVersionsCommand extends BaseCommand<typeof LatestVersionsCommand> {
+	static readonly summary =
+		"Determines if an input version matches a latest minor release version.";
+
+	static readonly description =
+		"This command is used in CI to determine if a pipeline was triggered by a release branch with the latest minor version of a major version.";
+
+	static readonly args = {
+		version: Args.string({
+			required: true,
+			description: "The version corresponding to the pipeline trigger branch.",
+		}),
+		package_or_release_group: packageOrReleaseGroupArg(),
+	} as const;
+
+	public async run(): Promise<boolean> {
+		const { args } = this;
+		const context = await this.getContext();
+		const versionInput = this.args.version;
+
+		if (args.package_or_release_group === undefined) {
+			this.error("No dependency provided.");
+		}
+
+		const rgOrPackage = findPackageOrReleaseGroup(args.package_or_release_group, context);
+		if (rgOrPackage === undefined) {
+			this.error(`Package not found: ${args.package_or_release_group}`);
+		}
+
+		const versions = await context.getAllVersions(rgOrPackage.name);
+
+		if (versions === undefined) {
+			this.error(`No versions found for ${rgOrPackage.name}`);
+		}
+
+		const sortedByVersion = sortVersions(versions, "version");
+
+		// Filter out versions that are not latest semver versions
+		const filteredVersions = sortedByVersion.filter((item) =>
+			/^\d+\.\d+\.\d+$/.test(item.version),
+		);
```

Comment:
In general, when working with versions, you should prefer to use the semver lib and parse them as semver, then work with the API to do whatever checks are needed. It's not very clear what you're trying to filter. There's only one "latest" semver version - so I'm guessing you mean something else?

Reply:
the semver lib makes it easy to sort a list of versions, so consider how you might improve this and the code below with a sorted list of versions that you could iterate over.

Reply:
The comment wasn't very clear but I meant to filter out the rc and internal versions from the list.
Updated filter to use semver.valid and semver.prerelease to filter out the versions instead of using regex.